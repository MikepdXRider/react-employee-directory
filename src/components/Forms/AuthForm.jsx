import React, { useState } from 'react'
import { useHistory } from 'react-router-dom';
import useAuth from '../../hooks/useAuth.jsx';
import { signInUser, signUpUser } from '../../services/users.js';

export default function AuthForm({isSigningUp=false}) {
    const history = useHistory();
    const [emailInput, setEmailInput] = useState('');
    const [passwordInput, setPasswordInput] = useState('');
    const [error, setError] = useState('');
    const { setUser } = useAuth();

    async function handleSubmit(e){
        e.preventDefault();
        if (passwordInput.length < 8) setError('Password must be at least 8 characters long');

        try{
            if (isSigningUp) {
                const currentUser = await signUpUser(emailInput, passwordInput);
                setUser(currentUser);
                history.replace('/create-profile');
            } else {
                const currentUser= await signInUser(emailInput, passwordInput);
                setUser(currentUser);
                history.replace('/profile');
            }
        } catch(err) {
            console.log(err);
        }
    }

    return (
        <fieldset>
            <legend>{isSigningUp ? 'Sign-Up' : 'Sign-In'}</legend>
            <form onSubmit={handleSubmit}>
                <label htmlFor="email">Email</label>
                <input type="email" id="email" value={emailInput} onChange={(e) => setEmailInput(e.target.value)} required/>

                <label htmlFor="password">Password</label>
                <input type="password" id="password" value={passwordInput} onChange={(e) => setPasswordInput(e.target.value)} required/>

                <input aria-label='submit form button' type="submit" />
                
                {
                    error && <p>{error}</p>
                }
            </form>
        </fieldset>
    )
}
