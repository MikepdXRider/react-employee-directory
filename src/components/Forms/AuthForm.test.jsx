import { render } from '@testing-library/react'
import AuthForm from './AuthForm.jsx';

it('renders AuthForm', () => {
    const {container} = render(
            <AuthForm/>
    )

    expect(container).toMatchSnapshot();
})
