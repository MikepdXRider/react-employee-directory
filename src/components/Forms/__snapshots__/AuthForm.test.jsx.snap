// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AuthForm 1`] = `
<div>
  <form>
    <label
      for="email"
    >
      Email
    </label>
    <input
      id="email"
      required=""
      type="email"
      value=""
    />
    <label
      for="password"
    >
      Password
    </label>
    <input
      id="password"
      required=""
      type="password"
      value=""
    />
    <input
      aria-label="submit form button"
      type="submit"
    />
  </form>
</div>
`;
